From 8fc1f3b61b02320848b035ccccd59e04e77d3f6b Mon Sep 17 00:00:00 2001
From: Ivan Maidanski <ivmai@mail.ru>
Date: Wed, 24 Sep 2014 22:47:59 +0400
Subject: [PATCH] Workaround 'unknown attribute __alloc_size__' warning for
 Clang 3.5 (trunk)

* include/gc_config_macros.h (GC_ATTR_ALLOC_SIZE): Define to empty if
clang 3.5.0 (workaround "unknown attribute '__alloc_size__' ignored"
warning in clang3.5 (trunk)).
---
 include/gc_config_macros.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/gc_config_macros.h b/include/gc_config_macros.h
index db8a40d..13f7e77 100644
--- a/include/gc_config_macros.h
+++ b/include/gc_config_macros.h
@@ -245,7 +245,8 @@
 # if defined(__GNUC__) && (__GNUC__ > 4 \
         || (__GNUC__ == 4 && __GNUC_MINOR__ >= 3 && !defined(__ICC)) \
         || __clang_major__ > 3 \
-        || (__clang_major__ == 3 && __clang_minor__ >= 2))
+        || (__clang_major__ == 3 && __clang_minor__ >= 2 \
+            && (__clang_minor__ != 5 || __clang_patchlevel__ != 0)))
 #   define GC_ATTR_ALLOC_SIZE(argnum) __attribute__((__alloc_size__(argnum)))
 # else
 #   define GC_ATTR_ALLOC_SIZE(argnum)
